cmd/virt-dra-controller/driver.go

```diff
@@ -0,0 +1,309 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	resourcev1 "k8s.io/api/resource/v1alpha2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/dynamic-resource-allocation/controller"
+	pcicrd "kubevirt.io/kubevirt-dra-driver/api/kubevirt.io/resource/pci/v1alpha1"
+
+	nascrd "kubevirt.io/kubevirt-dra-driver/api/kubevirt.io/resource/pci/nas/v1alpha1"
+	nasclient "kubevirt.io/kubevirt-dra-driver/api/kubevirt.io/resource/pci/nas/v1alpha1/client"
+	clientset "kubevirt.io/kubevirt-dra-driver/pkg/kubevirt.io/resource/clientset/versioned"
+)
+
+const (
+	DriverAPIGroup = pcicrd.GroupName
+)
+
+type OnSuccessCallback func()
+
+type driver struct {
+	lock      *PerNodeMutex
+	namespace string
+	clientset clientset.Interface
+	pci       *pcidriver
+}
+
+var _ controller.Driver = &driver{}
+
+func NewDriver(config *Config) (*driver, error) {
+	return &driver{
+		lock:      NewPerNodeMutex(),
+		namespace: config.namespace,
+		clientset: config.clientSets.Example,
+		pci:       NewPciDriver(),
+	}, nil
+}
+
+func (d driver) GetClassParameters(ctx context.Context, class *resourcev1.ResourceClass) (interface{}, error) {
+	if class.ParametersRef == nil {
+		return pcicrd.DefaultDeviceClassParametersSpec(), nil
+	}
+	if class.ParametersRef.APIGroup != DriverAPIGroup {
+		return nil, fmt.Errorf("incorrect API group: %v", class.ParametersRef.APIGroup)
+	}
+	dc, err := d.clientset.PciV1alpha1().DeviceClassParameters().Get(ctx, class.ParametersRef.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("error getting PciClassParameters called '%v': %v", class.ParametersRef.Name, err)
+	}
+
+	return &dc.Spec, nil
+}
+
+func (d driver) GetClaimParameters(ctx context.Context, claim *resourcev1.ResourceClaim, class *resourcev1.ResourceClass, classParameters interface{}) (interface{}, error) {
+	if claim.Spec.ParametersRef == nil {
+		return pcicrd.DefaultPciClaimParametersSpec(), nil
```

Comment:
Why not returning nil here? The default PCI parameter looks wrong to me, like the device name shouldn't never be empty